src/blob-uploader.js

```diff
@@ -0,0 +1,102 @@
+// @flow
+'use strict'
+
+const privateVars = new WeakMap()
+
+function blobUploader (apiUrl /* :string */) {
+  if (!apiUrl) {
+    throw new TypeError('blobUploader expects a api URL during instantiation')
+  }
+  privateVars.set(this, {
+    uri: apiUrl
+  })
+}
+
+blobUploader.prototype.uploadBlob = function (
+  blob /*: Blob */
+) /* :Promise<number> */ {
+  if (!blob) {
+    return Promise.reject(new Error('blob argument not provided'))
+  }
+
+  if (!privateVars || !privateVars.get(this)) {
+    return Promise.reject(new Error('blobUploader uri not configured'))
+  }
```

Comment:
I think this check `!privateVars.get(this)` is being repeated on next `if` statement